debugger/src/main/scala/org.apache.daffodil.debugger.dap/Parse.scala

```diff
@@ -1117,11 +1152,13 @@ object Parse {
           pstate.mpstate.delimiters.toList.zipWithIndex.map { case (delimiter, i) =>
             Delimiter(if (i < pstate.mpstate.delimitersLocalIndexStack.top) "remote" else "local", delimiter)
           },
-          pstate.diagnostics
+          pstate.diagnostics,
+          infoset
         )
     }
     case class EndElement(state: StateForDebugger) extends Event
     case object Fini extends Event
+    case class Control(state: DAPodil.Debugee.State) extends Event
```

Comment:
Rather than having two queues, one for data (state) changes, and another for events (control), we have a single event type so we can emit & commit state changes before event notifications are delivered.